### PR TITLE
Prevent Stage0 Environment Variables from leaking into Stage2 during execution in MSBuild

### PR DESCRIPTION
**Pull request by piotrpMSFT**
@livarcocc @jonsequitur 

This will support #4214

### CODE REVIEW

File: build/Microsoft.DotNet.Cli.Test.targets [hunk 1]
```diff
@@ -101,4 +102,9 @@
                 MsbuildArgs="%(TestPackageProject.MsbuildArgs) /p:SdkNuGetVersion=$(SdkNugetVersion)" />
 
   </Target>
+
+  <Target Name="EnsureStageSeparation">
+    <DotNetMSBuild Arguments="  /v:diag $(RepoRoot)/build_projects/Microsoft.DotNet.Cli.Build.SelfTest/InvokeWithStage0.proj /p:Stage2Directory=&quot;$(Stage2Directory)&quot;"
```
Comment (livarcocc) [inline build/Microsoft.DotNet.Cli.Test.targets]: Do you need the empty space at the beginning of Arguments?

Comment (piotrpMSFT) [inline build/Microsoft.DotNet.Cli.Test.targets]: fixed

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Windows_NT x86 Debug Build please.

**Comment by piotrpMSFT** [comment]
@rainersigwald @andygerlicher @cdmihai FYI, this is our implementation atop microsoft/msbuild#1610

**Comment by piotrpMSFT** [comment]
@MattGertz FYI, approving this infrastructure change which explicitly prevents the outer MSBuild from impacting behavior of the CLI being built by cleaning out environment variables that were previously leaking across the boundary.

**Pull request merged**